src/themes/default/components/core/ProductImage.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <div>
+    <img
+      src="/assets/placeholder.svg"
+      v-show="showPlaceholder"
+      itemprop="image"
+      key="placeholderImage"
+      ref="images"
+      class="mw-100 inline-flex">
+    <img
+      :src="images.loading"
+      :alt="alt"
+      v-if="!lowerQualityImageError || isOnline"
+      v-show="showLowerQualityImage"
+      @load="lowerQualityImageLoaded(true)"
+      @error="lowerQualityImageLoaded(false)"
+      itemprop="image"
+      key="lowQualityImage"
+      ref="images"
+      class="mw-100 inline-flex">
+    <img
+      :src="images.src"
+      :alt="alt"
+      v-if="!highQualityImageError || isOnline"
+      v-show="showHighQualityImage"
+      @load="highQualityImageLoaded(true)"
+      @error="highQualityImageLoaded(false)"
+      itemprop="image"
+      key="highQualityImage"
+      ref="images"
+      class="mw-100 inline-flex">
+  </div>
+</template>
+<script>
+import { onlineHelper } from '@vue-storefront/core/helpers'
+export default {
+  props: {
+    images: {
+      type: Object,
+      required: true
+    },
+    alt: {
+      type: String,
+      required: true
```

Comment:
```suggestion
      default: ''
```
we shouldn't use required in props unless it's completely necessary, otherwise we should provide default values